gpt4all-bindings/python/gpt4all/pyllmodel.py

```diff
@@ -254,26 +247,31 @@ def prompt_model(
         ----------
         prompt: str
             Question, task, or conversation for model to respond to
-        streaming: bool
-            Stream response to stdout
+        callback(token_id:int, response:str): bool
+            The model sends response tokens to callback
 
         Returns
         -------
-        Model response str
+        None
         """
 
-        prompt_bytes = prompt.encode('utf-8')
-        prompt_ptr = ctypes.c_char_p(prompt_bytes)
-
-        old_stdout = sys.stdout
+        if verbose:
+            print('====VERBOSE====')
+            print(prompt)
+            print('===/VERBOSE/===')
 
-        stream_processor = DualStreamProcessor()
 
-        if streaming:
-            stream_processor.stream = sys.stdout
+        prompt_bytes = prompt.encode('utf-8')
+        prompt_ptr = ctypes.c_char_p(prompt_bytes)
 
-        sys.stdout = stream_processor
+        def _callback_wrapper(callback):
+            
+            def _callback(token_id, response):
+                nonlocal callback
+                return callback(token_id, response.decode('utf-8', 'replace'))
```

Comment:
Ah! Here it is!

Right, I'll stop at this point with the review. The review now needs its own review I guess. 😅

I think we will need to talk about the whole Unicode thing before continuing like this.

Reply:
Since it's a separate topic of discussion and there is another PR about it (emojis PR: https://github.com/nomic-ai/gpt4all/pull/867) I leave this part as it is at this point. Maybe the emojis PR can be revived, or there should be a new PR with a solution and a discussion regarding this. If this _must_ be handled in this PR, then I can of course implement or copy the solution from the emojis PR.

Reply:
It's not necessary to be handled in this PR. I'll have another look at all the PR code sometime later, and I'll just need to see that it's still possible to extend that way.